library/common/extensions/filters/http/test_kv_store/BUILD

```diff
@@ -0,0 +1,45 @@
+load(
+    "@envoy//bazel:envoy_build_system.bzl",
+    "envoy_cc_extension",
+    "envoy_extension_package",
+    "envoy_proto_library",
+)
+
+licenses(["notice"])  # Apache 2
+
+envoy_extension_package()
+
+envoy_proto_library(
+    name = "filter",
+    srcs = ["filter.proto"],
+    deps = [
+        "@envoy_api//envoy/config/common/matcher/v3:pkg",
+    ],
+)
+
+envoy_cc_extension(
+    name = "test_kv_store_filter_lib",
+    srcs = ["filter.cc"],
+    hdrs = ["filter.h"],
+    repository = "@envoy",
+    deps = [
+        "filter_cc_proto",
+        "//library/common/api:c_types",
+        "//library/common/api:external_api_lib",
+        "//library/common/data:utility_lib",
+        "//library/common/extensions/key_value/platform:config",
+        "@envoy//source/common/common:assert_lib",
+        "@envoy//source/extensions/filters/http/common:pass_through_filter_lib",
+    ],
+)
+
+envoy_cc_extension(
+    name = "config",
+    srcs = ["config.cc"],
+    hdrs = ["config.h"],
+    repository = "@envoy",
+    deps = [
+        ":test_kv_store_filter_lib",
```

Comment:
Maybe worth a comment explaining why this isn't ideal but necessary right now to allow testing the interface?

Reply:
I did add this comment here:
https://github.com/envoyproxy/envoy-mobile/pull/2134/files#diff-f38766704cb1bb34da15e2496c577b3bd09a101137f0f2f684c6dc869821fe60R18

Or did were you thinking of something else? This is mostly a holdover from how things were set up historically and a desire to limit the scope of this PR. :)

Reply:
No need to add another comment, that one's sufficient. Thanks!